checks/redis/redis.go

```diff
@@ -0,0 +1,21 @@
+package redis
+
+import (
+	"context"
+	"errors"
+
+	"github.com/go-redis/redis"
+)
+
+// New creates a new Redis health check function.
+func New(client *redis.Client) func(ctx context.Context) error {
```

Comment:
Can we pass a connection string instead of a Redis client?

Reply:
I decided to pass a redis client, because there are a lot of connection options you can set for an client: https://github.com/go-redis/redis/blob/180f107add989d56d34734f1d429a83c6d4b9b91/options.go#L31-L130